### ISSUE TITLE
CLI - Added new commands for catalog for 1.6

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#12048]
```
# rancher catalog show --help
Show catalog template versions

Show all catalog template versions in the current $RANCHER_ENVIRONMENT. Use `--env <envID>` or `--env <envName>` to select a different environment.

Example:
  $ rancher --env k8slab catalog show <TEMPLATE_ID>

Usage: rancher [global options] show [OPTIONS] [TEMPLATE_ID]...

Options:
   --quiet, -q     Only display IDs
   --format value  'json' or Custom format: '{{.ID}} {{.Template.Id}}'
   --system, -s    Show system templates, not user
```

```
# rancher catalog upgrade --help
Upgrade stack with new catalog template version

Upgrade stack with new catalog template version in the current $RANCHER_ENVIRONMENT.
Use `--env <envID>` or `--env <envName>` to select a different environment.

Example:
  $ rancher --env k8slab catalog upgrade <TEMPLATE_VERSION_ID> --stack id

Usage: rancher [global options] upgrade [OPTIONS] [TEMPLATE_VERSION_ID]...

Options:
   --answers value, -a value  Answer file
   --stack value, -s value    Stack id to upgrade
   --confirm                  Wait for upgrade and confirm it
```

- Updated catalog list format, showing templates id's and default template version id.
```
# rancher catalog
ID                                  NAME                             CATEGORY                     VERSION                VERSION ID
community:Prometheus                Prometheus                       Monitoring                   3.0.0                  community:Prometheus:5
community:bitbucket                 bitbucket                        Versioning                   5.6.1                  community:bitbucket:1
community:pxc                       Percona XtraDB Cluster           Databases                    5.6.28-rancher1        community:pxc:0
community:plone                     Plone                            CMS                          5.0-rancher1           community:plone:1
community:adminer                   PHP Adminer                      Applications                 v1                     community:adminer:0
community:drupal                    Drupal                           CMS                          8.3.5-rancher1         community:drupal:1
community:kafka                     Apache Kafka                     Clustering                   1.0.0-rancher1         community:kafka:4
```
- Show all available catalog template versions and corresponding id of a template
```
# rancher catalog show community:traefik
VERSION                    ID
v1.0.0-beta.555-rancher1   community:traefik:0
v1.0.1-rancher1            community:traefik:1
v1.0.2-rancher1            community:traefik:2
v1.0.3-rancher1            community:traefik:3
v1.1.1-rancher1            community:traefik:4
v1.1.2-rancher1            community:traefik:5
v1.2.3-1-rancher1          community:traefik:6
v1.3.3-rancher1            community:traefik:7
v1.3.3-rancher2            community:traefik:8
v1.3.5-rancher1            community:traefik:9
v1.3.6-rancher1            community:traefik:10
v1.4.0-rancher1            community:traefik:11
v1.4.3-rancher1            community:traefik:12
v1.4.4-rancher1            community:traefik:13
v1.4.5-rancher1            community:traefik:14
v1.4.5-rancher2            community:traefik:15
v1.4.6-rancher1            community:traefik:16
```
- Upgrade stacks from catalog. Optional automatic confirm upgrade.
```
# rancher_darwin-amd64 catalog upgrade catalog://community:traefik:16 --stack 1st578 [--confirm]
```
- Show available upgrades for stacks deployed from catalog.
```
# rancher stack
ID        NAME                  STATE      CATALOG                           SERVICES   SYSTEM    DETAIL    AVAILABLE UPGRADES
1st332    gocd                  healthy    catalog://community:gocd:0        3          false
1st340    web-test2             healthy                                      2          false
1st397    web-microservice-12   degraded                                     5          false
1st464    zookeeper             healthy    catalog://community:zookeeper:4   1          false
1st465    kafka                 healthy    catalog://community:kafka:4       1          false
1st561    traefik               healthy    catalog://community:traefik:16    2          false
1st580    test                  healthy    catalog://community:traefik:15    1          false               community:traefik:16
```

**Comment by deniseschannon** [comment]
Available with v0.6.8-rc3

**Comment by soumyalj** [comment]
Tested with master using CLI v0.6.8-rc3. The above new commands were tested and are functional.

```
soumyas-MBP:rancher-v0.6.8-rc3 soumya$ ./rancher --env newcattle catalog show community:wordpress
ID                      VERSION
community:wordpress:0   v0.1-educaas1
community:wordpress:1   v0.2-bitnami

```
Display template ID and versions
```
soumyas-MBP:rancher-v0.6.8-rc3 soumya$ ./rancher catalog show library:k8s
ID               VERSION
library:k8s:0    v0.1.0-rancher1
library:k8s:1    v1.2.4-rancher6
library:k8s:2    v1.2.4-rancher6.1
library:k8s:3    v1.2.4-rancher7
library:k8s:5    v1.2.4-rancher7.2
library:k8s:6    v1.2.4-rancher7.3
library:k8s:7    v1.3.0-rancher3
library:k8s:8    v1.2.4-rancher8
library:k8s:9    v1.2.4-rancher9
library:k8s:10   v1.3.0-rancher4
library:k8s:11   v1.2.4-rancher10
```



*Available upgrades for stacks deployed from catalog:
```
soumyas-MBP:rancher-v0.6.8-rc3 soumya$ ./rancher stack
ID        NAME              STATE     CATALOG                               SERVICES   SYSTEM    DETAIL    AVAILABLE UPGRADES
1st6      teststackrename   healthy                                         1          false               
1st12     wordpress         healthy   catalog://community:wordpress:0       2          false               community:wordpress:1
1st13     elasticsearch     healthy   catalog://community:elasticsearch:0   4          false               community:elasticsearch:1
```
The upgrade for word press and elastic search catalogs succeeded with the commands below:

```
./rancher catalog upgrade community:wordpress:1 --stack 1st12 --confirm
./rancher catalog upgrade community:elasticsearch:1 --stack 1st13 --confirm

```
